pkg/base/helm.go

```diff
@@ -271,3 +274,104 @@ func checkChartForVersion(file *upstreamtypes.UpstreamFile) (string, error) {
 	// if no determination is made, assume v2
 	return "v2", nil
 }
+
+// insert namespace if it's defined in the spec and not already present in the manifests
+func kustomizeHelmNamespace(baseFiles []BaseFile, renderOptions *RenderOptions) ([]BaseFile, error) {
+	if renderOptions.Namespace == "" {
+		return baseFiles, nil
+	}
+
+	chartsPath, err := ioutil.TempDir("", "charts")
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to create temp dir")
+	}
+	defer os.RemoveAll(chartsPath)
+
+	var updatedBaseFiles []BaseFile
+	var kustomizeResources []string
+	var kustomizePatches []kustomizetypes.PatchStrategicMerge
+	resources := map[string]BaseFile{}
+	foundGVKNamesMap := map[string]bool{}
+	for _, baseFile := range baseFiles {
+		// write temp files for manifests that need a namespace
+		gvk, manifest := GetGVKWithNameAndNs(baseFile.Content, renderOptions.Namespace)
+		if manifest.APIVersion == "" && manifest.Kind == "" && manifest.Metadata.Name == "" {
```

Comment:
i feel like this should be or'ed instead of and'ed

Reply:
done